pub/assets/js/hound.js

```diff
@@ -284,12 +284,13 @@ var Model = {
   UrlToRepo: function(repo, path, line) {
     var info = this.repos[repo],
         url = info.url.replace(/\.git$/, ''),
-        anc = line ? '#L' + line : '';
+        anc = line ? info.anchorpattern.replace('${line}', line) : '';
 
     // Hacky solution to fix _some_ of the 404's when using SSH style URLs
     url = url.replace("git@github.com:", 'https://www.github.com/');
```

Comment:
Regarding "if we are going this way": The clean way would be to indicate a repository layout and define the repository layout elsewhere. So the JSON config would look like:

    {
        "dbpath" : "data",
        "repos" : {
            "SomeGitRepo" : {
                "url" : "https://www.github.com/YourOrganization/RepoOne.git",
                "layout" : "github"
            },
            "SomeStashRepo" : {
                "url" : "https://www.stashdomain.tld/scm/YourOrganization/RepoTwo.git",
                "layout" : "stash",
                "baseUrl" : "https://www.stashdomain.tld",
                "project" : "MyProject",
                "repoName" : "RepoTwo"
            }
        },
        "repositoryLayouts" : {
            "git" : {
                "anchorpattern" : "#L${line}",
                "urlpattern" : "${url}/blob/master/${path}${anc}"
            },
            "stash" : {
                "urlpattern" : "${baseUrl}/projects/${project}/repos/${repoName}/${path}${anc}",
                "anchorpattern" : "#${line}"
            }
        }
    }


But this leads to a configuration complexity, because starting with stash, there are at least two properties more to define for each repository. It will get even more complicated with personal repositories in stash, where the URL pattern is:

    https://stashdomain.tld/users/${MyUserName}/repos/${MyRepo}/browse/${path}${anc}

While the scm URL is:

    https://stashdomain.tld/scm/~${MyUserName}/${MyRepo}.git

So I'd prefer the url pattern configuration per repository.